src/components/Nav/Nav.js

```diff
@@ -0,0 +1,125 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import UserMenu from './components/UserMenu';
+import variables from '../../styles/variables';
+
+const NavContainer = styled.div`
+  ${variables.flex('row', 'space-between', 'center')}
+  position: sticky;
+  top: 0;
+  height: 80px;
+  padding: 0 80px;
+  border-bottom: 1px solid #e8e8e8;
+  background-color: white;
+`;
+
+const NavLogoBox = styled.div`
+  ${variables.flex('row', null, 'center')}
+  height: 80px;
+  cursor: pointer;
+  p {
+    font-size: 26px;
+    font-weight: bold;
+    color: ${props => props.theme.RedButton};
+  }
+`;
+
+const NavFilterBox = styled.div`
+  ${variables.flex('row', 'space-between', 'center')}
+  padding: 0 10px;
+  width: 250px;
+  height: 46px;
+  border-radius: 30px;
+  border: 1px solid #e8e8e8;
+  box-shadow: 0px 6px 6px -3px rgba(88, 88, 88, 0.29);
+
+  &:hover {
+    transition: 0.2s;
+    box-shadow: 0px 8px 9px -4px rgba(94, 94, 94, 0.5);
+  }
+  &:not(hover) {
+    transition: 0.5s;
+  }
+  p {
+    padding-left: 30px;
+    font-size: 13px;
+    font-weight: 600;
+  }
+`;
+
+const UserLoginSetting = styled.div`
+  ${variables.flex('row', 'space-around', 'center')}
+  padding: 0 10px;
+  width: 65px;
+  height: 33px;
+  padding: 5px 5px 5px 12px;
+  border-radius: 30px;
+  border: 1px solid #e8e8e8;
+  &:hover {
+    transition: 0.2s;
+    box-shadow: 0px 6px 6px -3px rgba(88, 88, 88, 0.29);
+  }
+  &:not(hover) {
+    transition: 0.5s;
+  }
+  div {
+    ${variables.flex('row', 'center', 'center')}
+    width: 29px;
+    height: 29px;
+    border-radius: 50px;
+    color: white;
+    background-color: #717171;
+    .fa-user {
+      font-size: 20px;
+    }
+  }
+  .fa-bars {
+    color: #717171;
+  }
+`;
+
+const UserModalBox = styled.div`
+  position: relative;
+`;
+
+const SearchIcon = styled.button`
+  width: 32px;
+  height: 32px;
+  border: none;
+  border-radius: 50px;
+  background-color: ${props => props.theme.RedButton};
```

Comment:
```suggestion
  background-color: ${(theme) => theme.RedButton};
```
구조분해할당을 하면 좀 더 깔끔하고 가독성을 높일 수 있을 것 같아요.